gravity.sh

```diff
@@ -1,476 +1,626 @@
 }
 
-gravity_advanced() {
-  # Remove comments and print only the domain name
-  # Most of the lists downloaded are already in hosts file format but the spacing/formating is not contigious
-  # This helps with that and makes it easier to read
-  # It also helps with debugging so each stage of the script can be researched more in depth
-  local str="Formatting list of domains to remove comments"
+# Clean up after Gravity
+gravity_Cleanup() {
+  local error="${1:-}"
+
+  str="Cleaning up stray matter"
   echo -ne "  ${INFO} ${str}..."
 
-  #awk '($1 !~ /^#/) { if (NF>1) {print $2} else {print $1}}' ${piholeDir}/${matterAndLight} | sed -nr -e 's/\.{2,}/./g' -e '/\./p' >  ${piholeDir}/${supernova}
-  #Above line does not correctly grab domains where comment is on the same line (e.g 'addomain.com #comment')
-  #Awk -F splits on given IFS, we grab the right hand side (chops trailing #coments and /'s to grab the domain only.
-  #Last awk command takes non-commented lines and if they have 2 fields, take the left field (the domain) and leave
-  #+ the right (IP address), otherwise grab the single field.
-  cat ${piholeDir}/${matterAndLight} | \
-    awk -F '#' '{print $1}' | \
-    awk -F '/' '{print $1}' | \
-    awk '($1 !~ /^#/) { if (NF>1) {print $2} else {print $1}}' | \
-    sed -nr -e 's/\.{2,}/./g' -e '/\./p' >  ${piholeDir}/${supernova}
+  rm ${piholeDir}/pihole.*.txt 2> /dev/null
+  rm ${piholeDir}/*.tmp 2> /dev/null
+
+  # Remove any unused .domains files
+  for file in ${piholeDir}/*.${domainsExtension}; do
+    # If list is not in active array, then remove it
+    if [[ ! "${activeDomains[*]}" =~ "${file}" ]]; then
+      rm -f "${file}" 2> /dev/null || \
+        echo -e "  ${CROSS} Failed to remove ${file##*/}"
+    fi
+  done
 
   echo -e "${OVER}  ${TICK} ${str}"
+  
+  [[ -n "${error}" ]] && echo ""
 
-  numberOf=$(wc -l < ${piholeDir}/${supernova})
-  echo -e "  ${INFO} ${COL_LIGHT_BLUE}${numberOf}${COL_NC} domains being pulled in by gravity"
+  # Only restart DNS service if offline
+  if ! pidof dnsmasq &> /dev/null; then
+    "${PIHOLE_COMMAND}" restartdns
+  fi
 
-  gravity_unique
+  # Print Pi-hole status if an error occured
+  if [[ -n "${error}" ]]; then
+    "${PIHOLE_COMMAND}" status
+    exit 1
+  fi
 }
 
-gravity_reload() {
-  # Reload hosts file
-  echo ""
-  local str="Refreshing lists in dnsmasq"
-  echo -e "  ${INFO} ${str}..."
-
-  # Ensure /etc/dnsmasq.d/01-pihole.conf is pointing at the correct list!
-  # First escape forward slashes in the path:
-  adList=${adList//\//\\\/}
-  # Now replace the line in dnsmasq file
-  #  sed -i "s/^addn-hosts.*/addn-hosts=$adList/" /etc/dnsmasq.d/01-pihole.conf
+helpFunc() {
+  echo "Usage: pihole -g
+Update domains from blocklists specified in adlists.list
 
-  "${PIHOLE_COMMAND}" restartdns
+Options:
+  -f, --force          Force the download of all specified blocklists
+  -h, --help           Show this help dialog"
```

Comment:
Should some of the below options be added to this list?

Reply:
I don't follow?

Reply:
There's several undocumented flags below which users might find helpful such as `--skip-download`

Reply:
Ah! I've definitely considered adding those flags, but opted not to because I couldn't think of any use case where they'd be needed.